### PR TITLE
check if valid value to set RepetingField as defined. fixed yearsuffi…

### PR DESCRIPTION
**Pull request by EgoLaparra**
…x equal to 0. fixed truncate weeks for first days of the year.

### CODE REVIEW

File: src/main/scala/info/bethard/timenorm/formal/Types.scala [hunk 1]
```diff
@@ -233,7 +233,10 @@ case class Year(digits: Int, nMissingDigits: Int = 0) extends Interval {
   */
 case class YearSuffix(interval: Interval, lastDigits: Int, nMissingDigits: Int = 0) extends Interval {
   val isDefined: Boolean = interval.isDefined
-  val nSuffixDigits: Int = (math.log10(lastDigits) + 1).toInt
+  val nSuffixDigits: Int = lastDigits match {
```
Comment (bethard) [inline src/main/scala/info/bethard/timenorm/formal/Types.scala]: Please add a test for this.

File: src/main/scala/info/bethard/timenorm/formal/Types.scala [hunk 2]
```diff
@@ -619,8 +622,7 @@ private[formal] object RepeatingInterval {
     case IsoFields.QUARTER_YEARS =>
       ldt.withMonth((ldt.getMonthValue - 1) / 4 + 1).withDayOfMonth(1).truncatedTo(ChronoUnit.DAYS)
     case ChronoUnit.MONTHS => ldt.withDayOfMonth(1).truncatedTo(ChronoUnit.DAYS)
-    case ChronoUnit.WEEKS =>
-      ldt.withDayOfYear(ldt.getDayOfYear - ldt.getDayOfWeek.getValue + 1).truncatedTo(ChronoUnit.DAYS)
+    case ChronoUnit.WEEKS => ldt.withDayOfYear((ldt.getDayOfYear - ldt.getDayOfWeek.getValue + 1) max 1).truncatedTo(ChronoUnit.DAYS)
```
Comment (bethard) [inline src/main/scala/info/bethard/timenorm/formal/Types.scala]: Please add a test for this. Make sure, for example, that the week of Tuesday January 1st is truncated to Monday December 31st of the preceding year.

Comment (EgoLaparra) [reply src/main/scala/info/bethard/timenorm/formal/Types.scala]: I have added a test and it seems to work well.

File: src/main/scala/info/bethard/timenorm/formal/Types.scala (hunk 1 above)
Comment (EgoLaparra) [reply src/main/scala/info/bethard/timenorm/formal/Types.scala]: I have found that YearSuffix didn't work when a Two-Digit-Year had leading zeros. I have made some changes to deal with this and also added a test.

File: src/main/scala/info/bethard/timenorm/formal/Types.scala [hunk 3]
```diff
@@ -231,9 +231,15 @@ case class Year(digits: Int, nMissingDigits: Int = 0) extends Interval {
   * As with Year, the optional second parameter allows YearSuffix to represent decades (nMissingDigits=1),
   * centuries (nMissingDigits=2), etc.
   */
-case class YearSuffix(interval: Interval, lastDigits: Int, nMissingDigits: Int = 0) extends Interval {
+case class YearSuffix(interval: Interval, lastDigits: Int, nlastDigits: Int = 0, nMissingDigits: Int = 0) extends Interval {
```
Comment (bethard) [inline src/main/scala/info/bethard/timenorm/formal/Types.scala]: I gather the problem here is that the calculation `nSuffixDigits` is doing to infer the number of digits that needs to be replaced is wrong with leading zeros. But why not just make `nSuffixDigits` a (non-optional) parameter and force everyone to supply that? It seems like that's what `nlastDigits` is trying to do in a roundabout way.

Comment (EgoLaparra) [reply src/main/scala/info/bethard/timenorm/formal/Types.scala]: Ok, much simpler. I was trying to keep the calculation of nSuffixDigits just in case.

**Pull request merged**